instance_config_interfaces.go

```diff
@@ -24,19 +44,61 @@ type VPCIPv4 struct {
+	Label       string                                    `json:"label,omitempty"`
+	Purpose     ConfigInterfacePurpose                    `json:"purpose,omitempty"`
+	Primary     bool                                      `json:"primary,omitempty"`
+	SubnetID    *int                                      `json:"subnet_id,omitempty"`
+	IPv4        *VPCIPv4                                  `json:"ipv4,omitempty"`
+	IPv6        *InstanceConfigInterfaceCreateOptionsIPv6 `json:"ipv6,omitempty"`
+	IPRanges    []string                                  `json:"ip_ranges,omitempty"`
+}
+
+// InstanceConfigInterfaceCreateOptionsIPv6 represents the IPv6 configuration of a Linode interface
+// specified during creation.
+type InstanceConfigInterfaceCreateOptionsIPv6 struct {
+	SLAAC    []InstanceConfigInterfaceCreateOptionsIPv6SLAAC `json:"slaac,omitempty"`
+	Ranges   []InstanceConfigInterfaceCreateOptionsIPv6Range `json:"ranges,omitempty"`
+	IsPublic *bool                                           `json:"is_public,omitempty"`
+}
+
+// InstanceConfigInterfaceCreateOptionsIPv6SLAAC represents a single IPv6 SLAAC of a Linode interface
+// specified during creation.
+type InstanceConfigInterfaceCreateOptionsIPv6SLAAC struct {
+	Range string `json:"range"`
+}
+
+// InstanceConfigInterfaceCreateOptionsIPv6Range represents a single IPv6 ranges of a Linode interface
+// specified during creation.
+type InstanceConfigInterfaceCreateOptionsIPv6Range struct {
+	Range *string `json:"range,omitempty"`
 }
 
 type InstanceConfigInterfaceUpdateOptions struct {
-	Primary  bool      `json:"primary,omitempty"`
-	IPv4     *VPCIPv4  `json:"ipv4,omitempty"`
-	IPRanges *[]string `json:"ip_ranges,omitempty"`
+	Primary  bool                                      `json:"primary,omitempty"`
+	IPv4     *VPCIPv4                                  `json:"ipv4,omitempty"`
+	IPv6     *InstanceConfigInterfaceUpdateOptionsIPv6 `json:"ipv6,omitempty"`
+	IPRanges *[]string                                 `json:"ip_ranges,omitempty"`
+}
+
+// InstanceConfigInterfaceUpdateOptionsIPv6 represents the IPv6 configuration of a Linode interface
+// specified during updates.
+type InstanceConfigInterfaceUpdateOptionsIPv6 struct {
+	SLAAC    *[]InstanceConfigInterfaceUpdateOptionsIPv6SLAAC `json:"slaac,omitempty"`
+	Ranges   *[]InstanceConfigInterfaceUpdateOptionsIPv6Range `json:"ranges,omitempty"`
+	IsPublic *bool                                            `json:"is_public,omitempty"`
+}
+
+// InstanceConfigInterfaceUpdateOptionsIPv6SLAAC represents a single IPv6 SLAAC of a Linode interface
+// specified during updates.
+type InstanceConfigInterfaceUpdateOptionsIPv6SLAAC struct {
+	Range *string `json:"range,omitempty"`
+}
+
+// InstanceConfigInterfaceUpdateOptionsIPv6Range represents a single IPv6 ranges of a Linode interface
+// specified during updates.
+type InstanceConfigInterfaceUpdateOptionsIPv6Range struct {
+	Range *string `json:"range,omitempty"`
 }
```

Comment:
I'm not a fan of the redundancy here but I wanted to make sure we can account for any potential future divergences.